net/dial/option.go

```diff
@@ -175,7 +175,7 @@ func WithTLSConfig(tlsConfig *tls.Config) DialOption {
 		}
 
 		do.afterHooks = append(do.afterHooks, AfterHook{
-			Priority: math.MaxUint64,
+			Priority: WithTLSConfigPriority,
```

Comment:
It is not recommended to change the default priority. 

A better approach would be to add a new function `WithTLSConfigAndPriority(tlsConfig *tls.Config, priority uint64)` that exports this functionality.